src/components/hoc/withTheme/withTheme.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { COLORS, TINTS } from '../../../constants';
+import PropsType from 'props-type';
+
+const props = {
+  /** The color of the component */
+  color: PropTypes.oneOf(COLORS),
+  /** The size of the component */
+  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'fullscreen']),
+  /** The tint of the component */
+  tint: PropTypes.oneOf(TINTS),
+  /** The css class of the component */
+  className: PropTypes.string,
+};
+
+const defaultProps = {
+  color: 'neutral',
+  size: 'medium',
+  tint: 'normal',
+};
+
+type Props = Partial<PropsType<typeof props, typeof defaultProps>>;
```

Comment:
I'm not entirely sure I'm following on why we need to use this library and can't just define the types in typescript itself 🤔

Reply:
Yes you can definitely do that! I used this library because this way you don't have to rewrite the proptypes so this should be easier to do.

Storybook also uses the proptypes to generate the documentation so we would have to use something like this to generate them again https://github.com/milesj/babel-plugin-typescript-to-proptypes
☝️ The resulting docs were not as 'good' though, the enums that show all options on hover just became the word `union` when I gave it a try.